scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,264 @@
+
+export const ${tokenDetails.symbol}_DECIMALS = ${tokenDetails.decimals};
+
+export const ${tokenDetails.symbol}_SYMBOL = '${tokenDetails.symbol}';
+
+export const ${tokenDetails.symbol}_TOKEN_ID: unique symbol = Symbol(${tokenDetails.symbol}_SYMBOL);
+
+export const ${mainnetToken}: RequiredErc20Token = {
+	id: ${tokenDetails.symbol}_TOKEN_ID,
+	network: ETHEREUM_NETWORK,
+	standard: 'erc20',
+	category: 'default',
+	name: '${tokenDetails.name}',
+	symbol: '${tokenDetails.symbol}',
+	decimals: ${tokenDetails.symbol}_DECIMALS,
+	icon: ${tokenName.toLowerCase()},
+	address: '${contractAddress}',
+	exchange: 'erc20',
+	twinTokenSymbol: 'ck${tokenDetails.symbol}'
+};
+
+export const SEPOLIA_${tokenDetails.symbol}_SYMBOL = 'Sepolia${tokenDetails.symbol}';
+
+export const SEPOLIA_${tokenDetails.symbol}_TOKEN_ID: unique symbol = Symbol(SEPOLIA_${tokenDetails.symbol}_SYMBOL);
+
+export const ${testnetToken}: RequiredErc20Token = {
+	id: SEPOLIA_${tokenDetails.symbol}_TOKEN_ID,
+	network: SEPOLIA_NETWORK,
+	standard: 'erc20',
+	category: 'default',
+	name: '${tokenDetails.name}',
+	symbol: '${tokenDetails.symbol}',
+	decimals: ${tokenDetails.symbol}_DECIMALS,
+	icon: ${tokenName.toLowerCase()},
+	address: '${testnetContractAddress}',
+	exchange: 'erc20',
+	twinTokenSymbol: 'ckSepolia${tokenDetails.symbol}'
+};
+`;
+
+	await fs.writeFile(newFilePath, newContent);
+	console.log(`Created ${newFilePath}`);
+
+	return { newFileName, mainnetToken, testnetToken };
+};
+
+const updateTokensErc20Env = async (newFileName, mainnetToken, testnetToken) => {
+	const filePath = path.join(SRC_PATH, 'tokens.erc20.env.ts');
+	let content = await fs.readFile(filePath, 'utf8');
+
+	const mainnetRegex = /const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token\[] = \[([\s\S]*?)];/;
+	const sepoliaRegex = /const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token\[] = \[([\s\S]*?)];/;
+
+	const mainnetMatch = content.match(mainnetRegex);
+	const sepoliaMatch = content.match(sepoliaRegex);
+
+	if (mainnetMatch) {
+		const tokensList = mainnetMatch[1].trim();
+		const updatedTokensList = tokensList ? `${tokensList}, ${mainnetToken}` : mainnetToken;
+		content = content.replace(
+			mainnetRegex,
+			`const ERC20_TWIN_TOKENS_MAINNET: RequiredErc20Token[] = [${updatedTokensList}];`
+		);
+	}
+
+	if (sepoliaMatch) {
+		const tokensList = sepoliaMatch[1].trim();
+		const updatedTokensList = tokensList ? `${tokensList}, ${testnetToken}` : testnetToken;
+		content = content.replace(
+			sepoliaRegex,
+			`const ERC20_TWIN_TOKENS_SEPOLIA: RequiredErc20Token[] = [${updatedTokensList}];`
+		);
+	}
+
+	const importStatement = `import { ${mainnetToken}, ${testnetToken} } from '$env/${newFileName.replace('.ts', '')}';\n`;
+	content = importStatement + content;
+
+	await fs.writeFile(filePath, content);
+	console.log(`Updated ${filePath}`);
+};
+
+const fetchSupportedTokens = async (url) => {
+	try {
+		const html = await fetchHtml(url);
```

Comment:
Few notes:

- Why searching for tokens online when those Oisy is supporting are listed in `tokens.ckerc20.json`? This json file is updated on a weekly basis and can also be updated thanks to a script or manually in the CI.

- If we have to fetch those information online, reading an HTML file to extract the list of token seems unreliable. Moreover, to some extension given that the outcome should be processed manually, not really secure. Why not calling the ck orchestrator canisters to get the list of supported tokens?

But, again, we have a json file. Maybe I just do not understand what the script is doing here?

Reply:
no, for sure it is better to use the json, I did that as first, but I thought that since it is inside `src` folder, you would have preffered to have on-way interactions between `scripts` --> `src`

i will use the json here

Reply:
> for sure it is better to use the json

Then let's just read the json file.

```
const obj = JSON.parse(readFileSync(pathToJson));
```

Reply:
Created PR #1505 to help getting the ERC20 contract address

Reply:
done